### ISSUE TITLE
sourcemap w/ empty arrays for `"sources"` and/or `"sourcesContent"` triggers error

### ISSUE THREAD

**Issue by calvinjuarez** [gulpjs/gulp#2313]
**What were you expecting to happen?**
Sourcemaps should work.  Specifically a source map with empty arrays for "sources" and/or "sourcesContent" should be treated like a missing sourcemap.

**What actually happened?**
Gulp chokes on a source map with empty arrays for `"sources"` and/or `"sourcesContent"`.

**Please post a sample of your gulpfile (preferably reduced to just the bit that's not working)**

```js
function emptyMap() {
	return src(EMPTY_MAP, { sourcemaps: true })
		.pipe(dest('./processed', { sourcemaps: '.' }))
}
```

**What version of gulp are you using?**
CLI version 2.1.0
Local version 4.0.0

**What versions of npm and node are you using?**
npm: 6.9.0
node: 11.1.0

---

Here's a sample project that shows this failing.  It's got 3 tasks.
1. `missingMap`: Demonstrates that a file with a sourcemap comment pointing to a missing file runs just fine.
2. `emptyMap`: Demonstrates that a file with a sourcemap comment pointing to a file with a map with empty `"sources"` and `"sourcesContent"` arrays triggers an error.
3. `noty`: Demonstrates that a sourcemap in the wild (build by `grunt-sass`) is being generated with empty `"sources"` and `"sourcesContent"` arrays, which trigger the error.

[gulp-sourcemap-issue.zip](https://github.com/gulpjs/gulp/files/3024909/gulp-sourcemap-issue.zip)

**Comment by phated** [comment]
Can you create a repository instead of attaching a zip file?

**Comment by phated** [comment]
Also, what do you mean by "chokes on"? Is there an error or what?

**Comment by calvinjuarez** [comment]
Yeah, sorry, "chokes on" was meant to mean error.  Here's the repo: https://github.com/calvinjuarez/gulp-sourcemap-issue

**Comment by phated** [comment]
@calvinjuarez thanks for creating the repo for this and adding all the detail!  This was actually a big bug that I was able to trace this all the way into our now-and-later library.  I've now fixed it and published a v2.0.1 of that library.  If you remove your package-lock/node_modules and reinstall everything, it'll work - I tested it locally and it's all tasks are working the same.
